File: src/main/java/dev/zprestige/magnesium/mixins/MixinMinecraftClient.java
package dev.zprestige.magnesium.mixins;

import dev.zprestige.magnesium.Main;
import dev.zprestige.magnesium.event.impl.TickEvent;
import net.minecraft.client.MinecraftClient;
import org.spongepowered.asm.mixin.Mixin;
import org.spongepowered.asm.mixin.injection.At;
import org.spongepowered.asm.mixin.injection.Inject;
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo;

@Mixin(MinecraftClient.class)
public class MixinMinecraftClient {

    @Inject(method = "tick", at = @At("HEAD"))
    private void tick(CallbackInfo callbackInfo) {
        Main.Companion.getEventBus().post(new TickEvent());
    }
}

File: src/main/kotlin/dev/zprestige/magnesium/mixins/MixinMinecraftClient.kt
package dev.zprestige.magnesium.mixins

import dev.zprestige.magnesium.Main
import dev.zprestige.magnesium.event.impl.TickEvent
import net.minecraft.client.MinecraftClient
import org.spongepowered.asm.mixin.Mixin
import org.spongepowered.asm.mixin.injection.At
import org.spongepowered.asm.mixin.injection.Inject
import org.spongepowered.asm.mixin.injection.callback.CallbackInfo

@Mixin(MinecraftClient::class)
class MixinMinecraftClient {

    @Inject(method = ["tick"], at = [At(value = "HEAD")])
    private fun runTick(callbackInfo: CallbackInfo){
        Main.eventBus.post(TickEvent())
    }
}